timelock-impl/src/main/java/com/palantir/atlasdb/timelock/AsyncTimelockResource.java

```diff
@@ -128,7 +131,19 @@ public StartTransactionResponseV4 startTransactions(StartTransactionRequestV4 re
     @POST
     @Path("start-atlasdb-transaction-v5")
     public StartTransactionResponseV5 startTransactionsWithWatches(StartTransactionRequestV5 request) {
-        return timelock.startTransactionsWithWatches(request);
+        return unwrapListenableFuture(timelock.startTransactionsWithWatches(request));
+    }
+
+    private static <T> T unwrapListenableFuture(ListenableFuture<T> future) {
+        try {
+            return future.get();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RuntimeException(e);
+        } catch (ExecutionException e) {
+            Throwables.throwIfUncheckedException(e.getCause());
+            throw new RuntimeException(e.getCause());
```

Comment:
Again as before (except this time we get the cause if we reach here, which doesn't seem an intentional dif)